diesel/src/query_builder/sql_query.rs

```diff
@@ -164,3 +178,85 @@ where
 }
 
 impl<Conn, Query, Value, ST> RunQueryDsl<Conn> for UncheckedBind<Query, Value, ST> {}
+
+#[must_use = "Queries are only executed when calling `load`, `get_result`, or similar."]
+/// See `SqlQuery`
+pub struct BoxedSqlQuery<'f, DB: Backend, Inner> {
+    inner: Inner,
+    sql: String,
+    binds: Vec<Box<dyn Fn(AstPass<DB>) -> QueryResult<()> + 'f>>,
+}
+
+impl<DB: Backend, Inner> Debug for BoxedSqlQuery<'_, DB, Inner>
+where
+    for<'a> debug_query::DebugQuery<'a, Self, DB>: Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        query_builder::debug_query::<DB, _>(self).fmt(f)
+    }
+}
+
+impl<'f, DB: Backend, Inner> BoxedSqlQuery<'f, DB, Inner> {
+    pub(crate) fn new(inner: Inner) -> Self {
+        BoxedSqlQuery {
+            inner,
+            sql: "".to_string(),
+            binds: vec![],
+        }
+    }
+
+    /// See `SqlQuery`'s `bind`
+    pub fn bind<BindSt, Value>(mut self, b: Value) -> Self
+    where
+        BindSt: QueryId,
+        DB: HasSqlType<BindSt>,
+        Value: ToSql<BindSt, DB> + 'f,
+    {
+        self.binds.push(Box::new(move |mut out| {
+            out.push_bind_param_value_only(&b)
+        }));
+        self
+    }
+
+    /// See `SqlQuery`'s `sql`
```

Comment:
`SqlQuery` has no `sql` function, because the whole query is provided at once. Therefore that function is not necessary and should be removed. (Same for the sql field, that also seems to be unused then.)

Reply:
I did not realize `SqlQuery` has no `sql` function, however, normally you can add extra sql via [sql](file:///home/gentz/Documents/0ad/source/tools/rdb/diesel/target/doc/diesel/expression/struct.SqlLiteral.html#method.sql).

As far as I can tell, an `sql` function in `BoxedSqlQuery` is necessary for doing things like this in a loop:
```
                        replays_query = replays_query
                            .sql("`")
                            .sql(&id_str)
                            .sql("`.`")
                            .sql(id_mem)
                            .sql("`")
                            .sql(op)
                            .sql(match equal {
                                OrEqual::True => "=",
                                OrEqual::False => "",
                            })
                            .sql("? ")
                            .bind::<sql_types::Binary, _>(val)
```

I guess we could loop twice, once for compiling the sql query, and another for adding the binds, however, that strikes me as unergonomic.

Reply:
That example won't work. `SqlQuery` assumes that you manually write the whole sql query, including the bind parameters.

The `sql` function is something entirely different. (Though this function needs also be extended to support a variable number of joins, but that's another PR)

Reply:
This example does work, however, with `BoxedSqlQuery`.

Reply:
Should be noted that users can also do things like `diesel::dsl::sql::<sql_types::Integer>("Hello ").sql("World!");`

Reply:
I would like to hear the opinion of at least one other core team member on this. 
In my opinion the API should be symmetrical to `SqlQuery`, that means we need to add such an function there or remove it here.

Reply:
Made it symmetrical.